File: client/src/app/models/discussion.ts
import { Message } from './message';

export class Discussion {
    id: string;
    timestamp: number;
    user1: string;
    user2: string;
    unreadMessage: boolean;

    constructor(id: string, timestamp: number, user1: string, user2: string, unreadMessage: boolean) {
        this.id = id;
        this.timestamp = timestamp;
        this.user1 = user1;
        this.user2 = user2;
        this.unreadMessage = unreadMessage;
      }
  }

File: server/src/main/java/fr/mightycode/cpoo/server/model/Discussion.java
package fr.mightycode.cpoo.server.model;

import lombok.Data;

import java.util.UUID;

import jakarta.persistence.*;

@Data
@Entity
@Table(name = "discussions")
public class Discussion {

  @Id
  @GeneratedValue(strategy = GenerationType.UUID)
  private UUID id;

  @Column(name = "timestamp", nullable = false)
  private long timestamp;

  @Column(name = "user1", nullable = false)
  private String user1;

  @Column(name = "user2", nullable = false)
  private String user2;

  @Column(name = "unreadMessage", nullable = false)
  private boolean unreadMessage = false;

}
